lib/services/watcher.go

```diff
@@ -486,6 +530,7 @@ func (p *lockCollector) getResourcesAndUpdateCurrent(ctx context.Context) error
 	p.currentRW.Lock()
 	defer p.currentRW.Unlock()
 	p.current = newCurrent
+	p.isStale = false
```

Comment:
`isStale` is `lockCollector`-specific and isn't part of the generic `resourceWatcher`.

Reply:
What I mean is, we set staleness via `notifyStale`, shouldn't we unset it via a similar `notifyNotStale` method? resourceWatcher is what decides when staleness starts and ends, after all.

Reply:
Staleness must be notified of because it's not associated with any data update (obviously). However, once we successfully fetch/refresh the data (which is what `getResourcesAndUpdateCurrent` is for) the fact the data isn't stale anymore is immediately implied. That's why `notifyNotStale` is not necessary and could arguably even contribute to inconsistent local state.

Reply:
Fair enough.